src/ws/cockpithandlers.c

```diff
@@ -567,6 +567,9 @@ handle_resource (CockpitHandlerData *data,
         {
           cockpit_channel_response_serve (service, headers, response, where,
                                           cockpit_web_response_get_path (response));
+          /* po.js is magic, it loads po.LL.js depending on current client language */
+          if (g_str_has_suffix (path, "/po.js"))
```

Comment:
I don't disagree with the logic, but this is the wrong place for it.

Let's rather put it in the bridge, somewhere near where cockpitpackages calls into `cockpit_web_response_negotiation()`, conditional on it actually having used the language name to find the file.

Reply:
Ack, will do, thanks for the hint! I disagree with the logic as well, as it's just a horrible hack around this caching maze + weird magic around loading "po.js". I hope we find a better solution still, but if we get stuck again (like we did for the last 5 years), we may have to bite the bullet.

I'd still like to hammer a bit on [changing the package checksum for a different locale](https://github.com/cockpit-project/cockpit/issues/8160#issuecomment-1061646262). This fails, and I don't yet completely understand why.

Reply:
Reworked to be in package_content(). That required quite a bit more work, but the initial hack at least showed that this does work in principle.